app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java

```diff
@@ -110,22 +124,30 @@ private void setupPetDetailsView() {
         mPetBreed.setText(mPet.getBreedFullName());
         mPetDescription.setText(mPet.getDescription());
         mShelterContactName.setText(mPet.getContactName());
-        mShelterContactPhone.setText(mPet.getContactPhone());
-        mShelterLocation.setText(mPet.getLocation());
-        mShelterEmail.setText(mPet.getContactEmail());
+  //      mShelterContactPhone.setText(mPet.getContactPhone());
+  //      mShelterLocation.setText(mPet.getLocation());
+        addUnderline(mShelterLocation,mPet.getLocation());
+        addUnderline(mShelterEmail, mPet.getContactEmail());
+        addUnderline(mShelterContactPhone,mPet.getContactPhone());
+//        mShelterEmail.setText(mPet.getContactEmail());
     }
 
+    private void addUnderline(TextView textView,String textString) {
+        SpannableString content = new SpannableString(textString);
+        content.setSpan(new UnderlineSpan(), 0, content.length(), 0);
+        textView.setText(content);
+    }
```

Comment:
it's a shame Android doesn't have a text style property for underline you could just set in the XML layout file :(.

Nicely done!